fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -256,24 +252,15 @@ public String getPath() {
      * @return
      * @throws RepositoryException
      */
-    private Iterator<Iterator<FedoraResource>> nodeToGoodChildren(final Node input) throws RepositoryException {
-        @SuppressWarnings("unchecked")
-        final Iterator<Node> allChildren = input.getNodes();
-        final Iterator<Node> children = filter(allChildren, nastyChildren.negate()::test);
-        return transform(children, (new Function<Node, Iterator<FedoraResource>>() {
-
-            @Override
-            public Iterator<FedoraResource> apply(final Node input) {
-                try {
-                    if (input.isNodeType(FEDORA_PAIRTREE)) {
-                        return concat(nodeToGoodChildren(input));
-                    }
-                    return singletonIterator(nodeToObjectBinaryConverter.convert(input));
-                } catch (final RepositoryException e) {
-                    throw new RepositoryRuntimeException(e);
+    @SuppressWarnings("unchecked")
+    private Stream<FedoraResource> nodeToGoodChildren(final Node input) throws RepositoryException {
```

Comment:
I should have looked more closely. The `input.getNodes()` expression will throw a `RepositoryException`, so the `throws` bit needs to be left in there.